src/js/components/Accordion/Accordion.js

```diff
@@ -7,85 +7,76 @@ import { AccordionContext } from './AccordionContext';
-    ) {
-      return { activeIndexes, activeIndex };
-    }
-
-    return null;
+const Accordion = ({
+  onActive,
+  multiple,
+  animate,
+  children,
+  messages,
+  activeIndex,
+  ...rest
+}) => {
+  const [activeIndexes, setActiveIndexes] = useState([]);
+  const [stateActiveIndex, setStateActiveIndex] = useState();
+
+  // Derived state from props
+  // https://reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops
+  const derivedActiveIndexes = activeAsArray(activeIndex) || [];
+  if (
+    (typeof activeIndex !== 'undefined' || activeIndex !== stateActiveIndex) &&
+    derivedActiveIndexes.join() !== activeIndexes.join()
+  ) {
+    setActiveIndexes(derivedActiveIndexes);
+    setStateActiveIndex(activeIndex);
   }
 
-  state = {
-    activeIndexes: [],
-  };
-
-  onPanelChange = index => {
-    const { activeIndexes } = this.state;
+  const onPanelChange = index => {
     let nextActiveIndexes = [...(activeIndexes || [])];
-    const { onActive, multiple } = this.props;
 
-    const activeIndex = nextActiveIndexes.indexOf(index);
-    if (activeIndex > -1) {
-      nextActiveIndexes.splice(activeIndex, 1);
+    const nextActiveIndex = nextActiveIndexes.indexOf(index);
+    if (nextActiveIndex > -1) {
+      nextActiveIndexes.splice(nextActiveIndex, 1);
     } else if (multiple) {
       nextActiveIndexes.push(index);
     } else {
       nextActiveIndexes = [index];
     }
 
-    this.setState({ activeIndexes: nextActiveIndexes }, () => {
-      if (onActive) {
-        onActive(nextActiveIndexes);
-      }
-    });
+    setActiveIndexes(nextActiveIndexes);
+    if (onActive) {
+      onActive(nextActiveIndexes);
+    }
   };
 
-  render() {
-    const { animate, children, messages, ...rest } = this.props;
-    const { activeIndexes } = this.state;
-
-    delete rest.onActive;
-
-    return (
-      <Box role="tablist" {...rest}>
-        {Children.toArray(children).map((panel, index) => (
-          <AccordionContext.Provider
-            key={`accordion-panel_${index + 0}`}
-            value={{
-              active: activeIndexes.indexOf(index) > -1,
-              animate,
-              messages,
-              onPanelChange: () => this.onPanelChange(index),
-            }}
-          >
-            {panel}
-          </AccordionContext.Provider>
-        ))}
-      </Box>
-    );
-  }
+  // eslint-disable-next-line no-param-reassign
+  delete rest.onActive;
+
+  return (
+    <Box role="tablist" {...rest}>
+      {Children.toArray(children).map((panel, index) => (
+        <AccordionContext.Provider
+          key={`accordion-panel_${index + 0}`}
+          value={{
+            active: activeIndexes.indexOf(index) > -1,
+            animate,
+            messages,
+            onPanelChange: () => onPanelChange(index),
+          }}
+        >
+          {panel}
+        </AccordionContext.Provider>
+      ))}
+    </Box>
+  );
 }
 
+Accordion.defaultProps = {
```

Comment:
Can we embed the defaultProps values as part of the props declaration on lines #10-#18

Reply:
I thought it was not the recommended way of doing it (idk why though) and just came to know it is indeed the preferred way. Thanks for letting me know. I just made the change and also rearranged the props alphabetically.